mirgecom/initializers.py

```diff
@@ -616,22 +609,14 @@ def exact_rhs(self, discr, q, t=0.0):
         momrhs = 0 * mom
 
         # process the species components independently
-        specrhs = None
-        if self._nspecies > 0:
-            lump_locs = make_obj_array([self._spec_centers[i] + loc_update
-                                       for i in range(self._nspecies)])
-            rel_poss = make_obj_array([nodes - lump_locs[i]
-                                       for i in range(self._nspecies)])
-            rdotvs = make_obj_array([np.dot(rel_poss[i], v)
-                                     for i in range(self._nspecies)])
-            r2s = make_obj_array([np.dot(rel_poss[i], rel_poss[i])
-                                  for i in range(self._nspecies)])
-            expterms = make_obj_array(
-                [self._spec_amplitudes[i] * actx.np.exp(- r2s[i])
-                 for i in range(self._nspecies)]
-            )
-            specrhs = make_obj_array([2 * expterms[i] * rdotvs[i]
-                                      for i in range(self._nspecies)])
+        # @MikeC: Guess I need to insert rho0 here somewhere after adding it above?
```

Comment:
Yes, it should just be rho0 * (original).  How does the test pass without that?

Reply:
The test only looks at `__call__` I think, not `exact_rhs`. (And the original test used `rho0 = 1`; I changed that in this PR.)

Reply:
There is a test down in `test_euler.py` (at least there should be) that should be checking the RHS. The test (as written before your latest change to put rho0 in) should have cause that one to fail if rho0 was not present (i.e. and not = 1.0).  rho0 should be specified by user, not hard-coded.

Reply:
I see now, the test is just letting rho0 = 1.0.  Should have let it be != 1.0 so that it would test it better.